pkg/userop/gas.go

```diff
@@ -0,0 +1,187 @@
+package userop
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/shopspring/decimal"
+)
+
+func getPolygonGasPrices(chainId *big.Int) (*big.Int, *big.Int, error) {
+	var resp *http.Response
+	var err error
+
+	switch {
+	case chainId.Cmp(big.NewInt(137)) == 0:
+		resp, err = http.Get("https://gasstation.polygon.technology/v2")
+	case chainId.Cmp(big.NewInt(80002)) == 0:
+		resp, err = http.Get("https://gasstation.polygon.technology/amoy")
+	default:
+		return nil, nil, fmt.Errorf("unsupported chain ID: %v", chainId)
+	}
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("error fetching data: %v", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error reading response body: %v", err)
+	}
+
+	var gasData struct {
+		Fast struct {
+			MaxPriorityFee float64 `json:"maxPriorityFee"`
+			MaxFee         float64 `json:"maxFee"`
+		} `json:"fast"`
+	}
+
+	err = json.Unmarshal(body, &gasData)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+
+	maxFeePerGas := decimal.NewFromFloat(gasData.Fast.MaxFee).Mul(decimal.NewFromInt(1e9)).BigInt()
+	maxPriorityFeePerGas := decimal.NewFromFloat(gasData.Fast.MaxPriorityFee).Mul(decimal.NewFromInt(1e9)).BigInt()
```

Comment:
Though optional, I suggest moving `decimal.NewFromInt(1e9)` into a separate variable to make those long lines shorter as this style often occurs in clearsync repo

Reply:
68cb8e64bff038323cf947fc2e017df1c8dafce5